### PR TITLE
feat: adding gradient_accumulate for run_clm_flax

### PR DESCRIPTION
**Pull request by pphuc25**
# What does this PR do?
I've added gradient accumulation for Flax file training, recognizing that resource constraints can be a limitation. Gradient accumulation assists people in training models with large batch sizes, mitigating these limitations.
I would like to cc dual @sanchit-gandhi and @ArthurZucker for review my PR

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Hey! I'm not really convinced that this is the best place to add this. A community blog + adding ressources for this somewhere might be better as the doc about exemples mentions:

> While we strive to present as many use cases as possible, the scripts in this folder are just examples. It is expected that they won’t work out-of-the box on your specific problem and that you will be required to change a few lines of code to adapt them to your needs. To help you with that, most of the examples fully expose the preprocessing of the data. This way, you can easily tweak them. 

the full script is not highly maintained and we want users to rather use most recent / shared examples.

**Comment by pphuc25** [comment]
thank you for your information

**Comment by sanchit-gandhi** [comment]
Thanks for opening this PR @pphuc25! To expand a bit on what @ArthurZucker mentioned, the Flax examples scripts are designed to be extremely minimalistic, i.e. contain the minimum level of complexity to perform a training run with the library models. This helps make them easy to build on and expand for custom use cases, since there is very little code and complexity to comprehend. To this effect, gradient accumulation is a bit too advanced for the Flax examples, since it won't be leveraged by all users. If you're interested, you ca use Optax MultiSteps for storing mini-batch updates until the final update step: https://optax.readthedocs.io/en/latest/gradient_accumulation.html#splitting-updates-for-one-batch-over-multiple-steps. I believe this is the recommended way for implementing gradient accumulation in Flax, and is one I've tried before with good success.

**Comment by pphuc25** [comment]
This explanation is well-done and provides valuable information. I appreciate it, @sanchit-gandhi. It was very informative for me. Thank you
